lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -574,15 +574,24 @@ bool isaDistributedLayout(Attribute layout) {
          layout.isa<SliceEncodingAttr>();
 }
 
-bool isSharedEncoding(Value value) {
+template <typename T>
+bool isEncoding(Value value) {
   auto type = value.getType();
   if (auto tensorType = type.dyn_cast<RankedTensorType>()) {
     auto encoding = tensorType.getEncoding();
-    return encoding && encoding.isa<triton::gpu::SharedEncodingAttr>();
+    return encoding && encoding.isa<T>();
   }
   return false;
 }
 
+bool isSharedEncoding(Value value) {
+  return isEncoding<triton::gpu::SharedEncodingAttr>(value);
+}
+
+bool isDotOperandEncoding(Value value) {
+  return isEncoding<triton::gpu::DotOperandEncodingAttr>(value);
+}
```

Comment:
Rename these to hasEncoding<T>, hasSharedEncoding, and hasDotOperandEncoding.  We aren't checking that the value *is* an encoding, but rather that it is a Tensor that *has* a particular encoding.